### DISCUSSION TITLE
Problem with ERROR while compiling ffmpeg-avutil.1.0.0~rc1

### DISCUSSION THREAD

**Discussion by voluburator** [savonet/liquidsoap#1921] [Installation]
Ubuntu 18.04.5 LTS (GNU/Linux 4.15.0-147-generic x86_64)
OCaml version 4.08.0
OPAM version 2.0

Can't build any version from ffmpeg. Compilation error appears ((((

⬇ retrieved mad.0.5.0 (https://opam.ocaml.org/cache)
[ERROR] The compilation of ffmpeg-avutil.1.0.0rc1 failed at "dune build -p ffmpeg-avutil -j 7 @install".
∗ installed mad.0.5.0
⬇ retrieved liquidsoap.2.0.0rc1 (https://opam.ocaml.org/cache)
⬇ retrieved ffmpeg-avcodec.1.0.0~rc1 (https://opam.ocaml.org/cache)
⬇ retrieved mm.0.7.1 (https://opam.ocaml.org/cache)
⬇ retrieved ogg.0.7.0 (https://opam.ocaml.org/cache)
⬇ retrieved theora.0.4.0 (https://opam.ocaml.org/cache)
∗ installed ogg.0.7.0
∗ installed theora.0.4.0
∗ installed mm.0.7.1

#=== ERROR while compiling ffmpeg-avutil.1.0.0~rc1 ============================#

context 2.1.0 | linux/x86_64 | ocaml-base-compiler.4.08.0 | https://opam.ocaml.org#c19a4dd2
path /.opam/4.08.0/.opam-switch/build/ffmpeg-avutil.1.0.0rc1
command ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ffmpeg-avutil -j 7 @install
exit-code 1
env-file ~/.opam/log/ffmpeg-avutil-11970-02c35b.env
output-file ~/.opam/log/ffmpeg-avutil-11970-02c35b.out
output
avutil_stubs.c:1641:13: error: request for member ‘sw_format’ in something not a structure or union
[...]
^~
avutil_stubs.c:1642:13: error: request for member ‘width’ in something not a structure or union
frames_ctx->width = Int_val(_width);
^~
avutil_stubs.c:1643:13: error: request for member ‘height’ in something not a structure or union
frames_ctx->height = Int_val(_height);
^~
avutil_stubs.c:1646:9: warning: implicit declaration of function ‘av_hwframe_ctx_init’; did you mean ‘av_frame_clone’? [-Wimplicit-function-declaration]
ret = av_hwframe_ctx_init(hw_frames_ref);
^~~~~~~~~~~~~~~~~~~
av_frame_clone
<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
┌─ The following actions failed
│ λ build ffmpeg-avutil 1.0.0~rc1
└─

**Comment by toots** [comment]
Hi!

It looks like your version of `ffmpeg` is too old. Any chance you could try with a more recent version? 

I believe that you need version `4.x`. There are some PPA for ubuntu here: https://launchpad.net/~savoury1/+archive/ubuntu/ffmpeg4?field.series_filter=bionic

You should try this:
```
sudo add-apt-repository ppa:savoury1/ffmpeg4
sudo apt-get update
sudo apt-get install ffmpeg
```
